test/include/test_util/storage_test_util.h

```diff
@@ -268,6 +270,54 @@ class StorageTestUtil {
     return memcmp(one.Content(), other.Content(), one.Size()) == 0;
   }
 
+  template <class RowType1, class RowType2>
+  static bool ProjectionListEqualShallowMatchSchema(const storage::BlockLayout &layout1, const RowType1 *const one,
+                                                    const storage::ProjectionMap &oid_map1,
+                                                    const storage::BlockLayout &layout2, const RowType2 *const other,
+                                                    const storage::ProjectionMap &oid_map2,
+                                                    const std::unordered_set<catalog::col_oid_t> &add_cols,
+                                                    const std::unordered_set<catalog::col_oid_t> &drop_cols) {
+    for (const auto &itr : oid_map1) {
+      auto oid1 = itr.first;
+      auto idx1 = itr.second;
+      if (drop_cols.find(oid1) == drop_cols.end()) {  // Column not dropped
+        auto idx2 = oid_map2.at(oid1);
+        storage::col_id_t one_id = one->ColumnIds()[idx1];
+
+        // Check that the two have the same content bit-wise
+        uint8_t attr_size = layout1.AttrSize(one_id);
+        const byte *one_content = one->AccessWithNullCheck(idx1);
+        const byte *other_content = other->AccessWithNullCheck(idx2);
+        // Either both are null or neither is null.
+        if (one_content == nullptr || other_content == nullptr) {
+          EXPECT_EQ(one_content, other_content);
+          if (one_content == other_content) continue;
+          return false;
```

Comment:
If they should be equal, why are you checking twice?

Reply:
The first `EXPECT_EQ` check is so that the `one_content != other_content` is reported when test fails. 
The second return false is so that the invocation will be reported as false. 

As a result, you have better test output where you know 2 projected rows across schema versions do not match, and you know what exactly causes them not to match. 
And in fact, other test utility functions are have this "double check".